src/main/java/org/synyx/urlaubsverwaltung/security/SessionConfiguration.java

```diff
@@ -0,0 +1,58 @@
+package org.synyx.urlaubsverwaltung.security;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.session.JdbcSessionDataSourceScriptDatabaseInitializer;
+import org.springframework.boot.autoconfigure.session.JdbcSessionProperties;
+import org.springframework.boot.autoconfigure.session.SessionProperties;
+import org.springframework.boot.autoconfigure.web.ServerProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.session.config.SessionRepositoryCustomizer;
+import org.springframework.session.jdbc.JdbcIndexedSessionRepository;
+import org.springframework.session.jdbc.config.annotation.SpringSessionDataSource;
+import org.springframework.session.jdbc.config.annotation.web.http.EnableJdbcHttpSession;
+import org.synyx.urlaubsverwaltung.tenancy.configuration.single.ConditionalOnSingleTenantMode;
+
+import javax.sql.DataSource;
+
+/**
+ * Configuration for Spring Session with JDBC backend.
+ * This configuration is required, path a way for other possible session implementations
+ * and disable the JDBC based one.
+ */
+@ConditionalOnSingleTenantMode
+@Configuration
+@EnableJdbcHttpSession
+@EnableConfigurationProperties(JdbcSessionProperties.class)
+public class SessionConfiguration {
```

Comment:
Warum benötigen wir die Klasse? Spring Session kann überschrieben werden in den properties. Sehe ich was nicht?

Reply:
gab in der vergangenheit probleme, wenn mehrere session implementierungen auf dem classpath lagen, eine option in den properties dazu habe ich von spring nicht entdeckt, magst du das mal verlinken?

ich teste das nochmal, eventuell hat sich mit dem upgrades der spring boot versionen der letzten monate etwas verbessert.